sdk/php/src/Command/EntrypointCommand.php

```diff
@@ -0,0 +1,224 @@
+                    );
+                }
+
+                $objectTypeDef = $objectTypeDef->withFunction($func);
+            }
+
+            $daggerModule = $daggerModule->withObject($objectTypeDef);
+        }
+
+        $functionCall->returnValue(new Dagger\Json(json_encode(
+            (string) $daggerModule->id()
+        )));
+
+        return Command::SUCCESS;
+    }
+
+    private function callFunctionOnParent(
+        OutputInterface $output,
+        Dagger\FunctionCall $functionCall,
+        string $parentName
+    ): int {
+        $errorOutput = $output instanceof ConsoleOutputInterface ?
+            $output->getErrorOutput() :
+            $output;
+
+        $className = "DaggerModule\\$parentName";
+        $functionName = $functionCall->name();
+        $class = new $className();
+
+        $args = $this->formatArguments(
+            $className,
+            $functionName,
+            json_decode(json_encode($functionCall->inputArgs()), true)
+        );
+
+        try {
+            $result = ($class)->$functionName(...$args);
+        } catch (QueryError $e) {
+            if (!isset($e->getErrorDetails()['extensions'])) {
+                throw $e;
+            }
+
+            $errorOutput->writeln($e->getMessage());
+            $output->writeln($e->getErrorDetails()['extensions']['stdout'] ?? '');
+            $errorOutput->writeln($e->getErrorDetails()['extensions']['stderr'] ?? '');
+
+            return $e->getErrorDetails()['extensions']['exitCode'] ??
+                Command::FAILURE;
+        }
+
+        if ($result instanceof Dagger\Client\IdAble) {
+            $result = (string) $result->id();
+        }
+
+        $functionCall->returnValue(new Dagger\Json(json_encode($result)));
+
+        return Command::SUCCESS;
+    }
+
+
+    private function getTypeDef(ListOfType|Type $type): TypeDef
+    {
+        $typeDef = DAG->typeDef()->withOptional($type->nullable);
+
+        switch ($type->typeDefKind) {
+            case TypeDefKind::BOOLEAN_KIND:
+            case TypeDefKind::INTEGER_KIND:
+            case TypeDefKind::STRING_KIND:
+            case TypeDefKind::SCALAR_KIND:
+            case TypeDefKind::VOID_KIND:
+                return $typeDef->withKind($type->typeDefKind);
+            case TypeDefKind::LIST_KIND:
+                return $typeDef->withListOf($this->getTypeDef($type->subtype));
+            case TypeDefKind::INTERFACE_KIND:
+                throw new RuntimeException(sprintf(
+                    'Currently cannot handle custom interfaces: %s',
+                    $type->name
+                ));
+            case TypeDefKind::OBJECT_KIND:
+                if ($type->isIdable()) {
+                    return $typeDef->withObject($type->getShortName());
+                }
+
+                throw new RuntimeException(sprintf(
+                    'Currently cannot handle custom classes: %s',
+                    $type->name
+                ));
```

Comment:
This currently only works for objects that are already in the API but it's easy to support other objects in the same module. At this point, it's just handling references like in the return type of a method that returns the object, so the `withObject` here just needs to know the name of the class, and to check if it's a `[#DaggerFunction]`.

Reply:
I'll mark it as resolved since it appears there's already a new PR for this.